include/odp/api/spec/shared_memory.h

```diff
@@ -243,6 +243,31 @@ odp_shm_t odp_shm_import(const char *remote_name,
  */
 void *odp_shm_addr(odp_shm_t shm);
 
+/**
+ * Shared memory block physical address
+ *
+ * Returns the number of contiguous physical memory segments belonging to the
+ * shared memory block. Outputs up to 'num' physical memory segment start
+ * addresses when the 'pa' array pointer is not NULL. Also, outputs up to 'num'
+ * physical memory segment lengths when the 'size' array pointer is not NULL.
+ *
+ * If return value is larger than 'num', there are more memory segments than the
+ * function was allowed to output. If return value (N) is less than 'num', only
+ * output array elements[0 ... N-1] have been written.
+ *
+ * @param      shm    Block handle
+ * @param[out] pa     Array of physical addresses for output or NULL
+ * @param[out] size   Array of physical memory segment lengths for output or
+ *                    NULL
+ * @param      num    Maximum number of physical addresses and memory segment
+ *                    lengths to output
+ *
+ * @return Number of physical addresses
+ * @retval 0 on failure
+ */
+uint32_t odp_shm_addr_pa(odp_shm_t shm, void *pa[], uint64_t size[],
+			 uint32_t num);
```

Comment:
Pointer and size could be combined into a struct (odp_shm_addr_info_t). That would allow backward compatible additions in the future. We have odp_shm_info() already, but that does not seem flexible enough for variable number of PA addresses per block.

uint32_t odp_shm_addr_info(odp_shm_t shm, odp_shm_addr_info_t[], uint32_t num);

Reply:
Makes sense. We could add e.g. iova address there in the future.

Reply:
`odp_shm_addr_pa()` removed in V2.